src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1226,16 +1229,47 @@ executePointsTo opts sc cc spec (LLVMPointsTo _loc ptr val) =
   do (_, ptr1) <- resolveSetupValue opts cc sc spec Crucible.PtrRepr ptr
      sym    <- Ov.getSymInterface
 
-     -- In case the types are different (from crucible_points_to_untyped)
-     -- then the load type should be determined by the rhs.
-     (memTy1, val1) <- resolveSetupValueLLVM opts cc sc spec val
-     storTy <- Crucible.toStorableType memTy1
-
      let memVar = Crucible.llvmMemVar $ (cc^.ccLLVMContext)
      let alignment = Crucible.noAlignment -- default to byte alignment (FIXME)
      mem  <- readGlobal memVar
-     mem' <- liftIO (Crucible.storeRaw sym mem ptr1 storTy alignment val1)
-     writeGlobal memVar mem'
+
+     -- In case the types are different (from crucible_points_to_untyped)
+     -- then the load type should be determined by the rhs.
+     m <- OM (use setupValueSub)
+     s <- OM (use termSub)
+     let tyenv = MS.csAllocations spec
+     let nameEnv = MS.csTypeNames spec
+     memTy <- liftIO $ typeOfSetupValue cc tyenv nameEnv val
+     storTy <- Crucible.toStorableType memTy
+     val' <- liftIO $ instantiateSetupValue sc s val
+
+     smt_array_memory_model_enabled <- liftIO $ W4.getOpt
+      =<< W4.getOptionSetting
+        enableSMTArrayMemoryModel
+        (W4.getConfiguration sym)
+
+     mem'' <- case val' of
+      SetupTerm tm
```

Comment:
This looks very similar to the code in `Builtins.LLVM`. Are they the same? Could we make a helper function for this?